### DISCUSSION TITLE
Allowing specific users to see/add images related to a specific site group?

### DISCUSSION THREAD

**Discussion by pv2b** [netbox-community/netbox#18000] [Q&A]
Hello!

I'm looking to use the permission system to allow access to a specific site group, and objects related to that. Unfortunately, I'm running into a bit of a snag when it comes to Image Attachments.

I can't figure out how to write a query to say "image attachments, that belong to a rack, that belongs to a site". Let alone to add a site group into the mix.

I should be something like this:

    {"object_type__model": "dcim.rack", "parent__site__group__name": "FOO"}

But that leaves me with this validation error:

> Invalid filter for <class 'extras.models.models.ImageAttachment'>: Field 'parent' does not generate an automatic reverse relation and therefore cannot be used for reverse querying. If it is a GenericForeignKey, consider adding a GenericRelation.

Is this simply a limitation of the current object model of Netbox, i.e. the database isn't set up to handle this type of query, or is there some magic incantation I can use here?

Thanks in advance.

**Comment by jbailey-bci** [comment]
Ran into this when working on some restricted permissions in my instance. From what I understand, this is because the "assigned object" is generic and could point to various models (ie. there isn't a "hard" reference to follow). I think there is a potential for workarounds, but it will likely require code changes in NetBox.

Ideally, I'd like to see some way to say "if user has permissions to parent object, they can see/modify/etc any associated images". Since images have to be associated with another object, I think that would be a reasonable ask.
